android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUINativePicker.java

```diff
@@ -41,7 +41,8 @@
 {
 	private static final String TAG = "TiUINativePicker";
 	private boolean firstSelectedFired = false;
-	private static int defaultTextColor = 0;
+	private static int defaultTextColor = -1;
+	private static boolean setDefaultColor = false;
```

Comment:
Ah, I see a problem. We should be setting the `defaultColor` regardless if `PROPERTY_COLOR` is defined. Otherwise, the initial text wont be set to the default text color unless an item with `PROPERTY_COLOR` is set first. And yes, good catch keep the Boolean 👍 

See if this works?

```Java
private static int defaultTextColor;
private static boolean setDefaultTextColor = false;
...
private void styleTextView(int position, TextView tv) {
...
if (!setDefaultTextColor) {
    defaultTextColor = tv.getCurrentTextColor();
    setDefaultTextColor = true;
}
if (rowProxy.hasProperty(TiC.PROPERTY_COLOR)) {
    final int color = TiConvert.toColor(rowProxy.getProperties(), TiC.PROPERTY_COLOR);
    tv.setTextColor(color);
} else {
    tv.setTextColor(defaultTextColor);
}
...
```